prboom2/src/gl_preprocess.c

```diff
@@ -695,6 +705,163 @@ static void gld_PrecalculateSector(int num)
   Z_Free(lineadded);
 }
 
+static dboolean gld_IsRealLine(sector_t* sector, line_t* line)
+{
+  int i;
+  int v1c;
+  int v2c;
+
+  // Obvious case -- at least one side isn't sector itself
+  if (line->frontsector != sector || line->backsector != sector)
+    return true;
+
+  // Distinguish a normal mid-sector linedef from a mapping trick by
+  // checking connectedness of vertices
+  v1c = v2c = 0;
+  for (i = 0; i < sector->linecount; ++i)
+  {
+    if (sector->lines[i] == line)
+      continue;
+    if (sector->lines[i]->v1 == line->v1 || sector->lines[i]->v2 == line->v1)
+      v1c++;
+    if (sector->lines[i]->v2 == line->v2 || sector->lines[i]->v2 == line->v2)
+      v2c++;
+  }
+
+  // Line is still real if:
+  // - It's isolated (no neighbors), i.e. it's floating mid-sector
+  // - It has more than 1 neighbor per vertex, i.e. it divides the sector
+  return (v1c == 0 && v2c == 0) || (v1c > 1 && v2c > 1);
```

Comment:
Should it just be ``v1c != 1 && v2c != 1`? I.e., what about a line that has more than one neighbor on one vertex and no neighbors on the other vertex?

Reply:
Yeah, good point